modules/storage-api/src/main/java/org/apache/ignite/internal/storage/engine/MvTableStorage.java

```diff
@@ -116,16 +116,18 @@ default IndexStorage getOrCreateIndex(int partitionId, UUID indexId) {
      * @throws StorageException If the given partition does not exist, or the given index does not exist or is not configured as a
      *         hash index.
      */
-    HashIndexStorage getOrCreateHashIndex(int partitionId, UUID indexId);
+    HashIndexStorage getOrCreateHashIndex(int partitionId, UUID indexId) throws StorageException;
 
     /**
      * Destroys the index under the given name and all data in it.
      *
      * <p>This method is a no-op if the index under the given name does not exist.
      *
+     * @param partitionId Partition ID.
      * @param indexId Index ID.
+     * @throws StorageException If the given partition does not exist, or fail destroying the index.
      */
-    CompletableFuture<Void> destroyIndex(UUID indexId);
+    CompletableFuture<Void> destroyIndex(int partitionId, UUID indexId) throws StorageException;
```

Comment:
Why did you add a partition id? What does it mean? I don't understand it.

Reply:
Fix it